src/web/src/components/CourseList.vue

```diff
@@ -84,13 +101,15 @@ export default {
      * @returns {Course[]}
      */
     filteredCourses() {
-      return this.courses.filter(({ date_start, date_end, department, title }) => {
+      return this.courses.filter(({ date_start, date_end, department, title, semester }) => {
         return (
           (!this.selectedSubsemester ||
             (this.selectedSubsemester.date_start.getTime() === date_start.getTime() &&
               this.selectedSubsemester.date_end.getTime() === date_end.getTime())) &&
           (!this.selectedDepartment || this.selectedDepartment === department) &&
-          (!this.textSearch || title.includes(this.textSearch.toUpperCase()))
+          (!this.textSearch || title.includes(this.textSearch.toUpperCase())) &&
+          (!this.selectedSemester || this.selectedSemester ===
```

Comment:
Seems that the default value for `selectedSemester` is not a falsey value so this actually breaks when there is no defaultSemester

Reply:
Fixed